hvplot/converter.py

```diff
@@ -1347,26 +1362,28 @@ def dataset(self, x=None, y=None, data=None):
     def heatmap(self, x, y, data=None):
         data = self.data if data is None else data
         opts = dict(plot=self._plot_opts, norm=self._norm_opts, style=self._style_opts)
+
         if not (x or y) or (x == 'columns' and y in ('index', data.index.name)):
             opts['plot']['labelled'] = []
             x, y = 'columns', 'index'
             data = (data.columns, data.index, data.values)
             z = ['value']
         else:
-            if not x: x = self.x or data.columns[0]
-            if not y: y = self.y or data.columns[1]
             z = self.kwds.get('C', [c for c in data.columns if c not in (x, y)][0])
             z = [z] + self.hover_cols
+            self.use_index = False
+            data, x, y = self._process_chart_args(data, x, y, single_y=True)
+
         redim = self._merge_redim({z[0]: self._dim_ranges['c']})
         hmap = HeatMap(data, [x, y], z, **self._relabel)
         if 'reduce_function' in self.kwds:
             hmap = hmap.aggregate(function=self.kwds['reduce_function'])
         return hmap.redim(**redim).opts(**opts)
 
     def hexbin(self, x, y, data=None):
-        data = self.data if data is None else data
-        if not x: x = data.columns[0]
-        if not y: y = data.columns[1]
+        self.use_index = False
```

Comment:
Should we handle these in some more general way?

Reply:
I actually think that we should let them use index, but I didn't want to change prior behavior.

Reply:
I'd be happy to change it though if you don't think it'll be too painful

Reply:
Can you explain the consequences?

Reply:
instead of x defaulting to the first column, it'd default to the index , and then y instead of defaulting to the second columns would default to the first.

Reply:
Let's leave that for another time, I'm not totally sure we should do that for 2D plots.

Reply:
Doesn't Pandas .plot() do it?

Reply:
Pandas plot does not have these plot types.

Reply:
I mean, doesn't Pandas .plot() use the index in that way, in general?  I think hvPlot should follow what Pandas does, even for new plot types, to be predictable and unsurprising.

Reply:
These types of plots require explicit x and y in pandas.plot